### ISSUE TITLE
Auth Refresh Tokens - To keep users logged in

### ISSUE THREAD

**Issue by yossi-shasho** [juicyllama/framework#199]
Implement refresh tokens so that users can stay logged in after their tokens expired (> ±12 hrs).

**Comment by yossi-shasho** [comment]
How it works:

Upon login, users receive a secure cookie containing a refresh token. When an access token expires, our new front-end interceptor requests a new one from the auth/refresh endpoint, also generating a new refresh token in the process.

To do:

- Introduce new environment variables for client apps to configure token behavior.
1. JWT_REFRESH_KEY - required
2. JWT_REFRESH_TOKEN_EXPIRY_DAYS - optional. Default 14
3. JWT_ACCESS_TOKEN_EXPIRY_MINUTES - optional. Default 60
- Optionally, consider storing refresh tokens in our database. This allows for additional security measures, such as detecting and reacting to potential misuse of stolen tokens, enhancing overall system security.

**Comment by andyslack** [comment]
Ahh it looks like you handle this automatically via the [interceptor.ts](https://github.com/juicyllama/framework/commit/1490e5f3614b7cc8f5e260718d587ad5eeb82215#diff-234776c6f61fe52456f57af8964e480b8e03b848c960afb50b6ecf7a6a390ac3) All good :)

**Comment by yossi-shasho** [comment]
@andyslack 
 Introducing token refresh logic directly into this hook might blur its focused purpose. By delegating the token refresh to a dedicated Axios interceptor, we not only adhere to the principle of single responsibility, enhancing code maintainability, but also keep this process transparent to the Vue app itself. The interceptor operates at the request level rather than the application level, silently updating tokens

**Comment by andyslack** [comment]
Hi @yossi-shasho Once done, can you do a commit to https://github.com/juicyllama/client-quickstart to demonstrate the changes required.

The team can then mirror these over to the other client projects.

Cheers in advanced.

**Comment by yossi-shasho** [comment]
@andyslack The changes in ths PR make the CORS support strict to only the FE url and nothing more. Does any client app actually need the old, full CORS support?

**Comment by yossi-shasho** [comment]
@andyslack I dont think so -- CORS only instructs browser to limit XHR requests to an API. it wont affect any server-to-server communication, nor client-to-3rd-parties communication

**Comment by andyslack** [comment]
Hi @yossi-shasho 

Do you think we can use some better error message when users fail to authenticate:



![Image](https://github.com/juicyllama/framework/assets/3747768/2db5dd25-ed97-4716-ae4a-3188ac6c0103)

**Comment by yossi-shasho** [comment]
@andyslack I'll take care of that

**Comment by yossi-shasho** [comment]
So far haven't been able to reproduce this. I'll keep an eye on it
